LoopKit/Persistence/PersistenceController.swift

```
@@ -192,3 +224,34 @@ extension PersistenceController: CustomDebugStringConvertible {
         ].joined(separator: "\n")
     }
 }
+
+
+// MARK: - Anchor store/fetch helpers
+
+extension PersistenceController {
+    func storeAnchor(_ anchor: HKQueryAnchor?, key: String) {
+        managedObjectContext.perform {
+            let encoded: Data?
+            if let anchor = anchor {
+                encoded = try? NSKeyedArchiver.archivedData(withRootObject: anchor, requiringSecureCoding: true)
```

[nit] This can silently fail.

---

Added a log; still want the same behavior.